homeassistant/components/slack/notify.py

```diff
@@ -144,6 +155,70 @@ def __init__(
         self._hass = hass
         self._client = client
         self._config = config
+        self._channel_id_cache: dict[str, str] = {}
+
+    async def _async_get_dm_channel_id(self, user_id: str) -> str | None:
+        """Get the DM channel ID for a user ID.
+
+        Opens a DM channel with the user if one doesn't exist.
+        """
+        try:
+            response = await self._client.conversations_open(users=[user_id])
+            if response["ok"]:
+                return cast(str, response["channel"]["id"])
+
+            _LOGGER.error(
+                "Failed to open DM with user %s: %s", user_id, response["error"]
+            )
+        except SlackApiError as err:
+            _LOGGER.error("Error opening DM with user %s: %r", user_id, err)
```

Comment:
This is swallowing the error instead of failing the action.
I think it is better to raise an appropriate error instead.